src/App.js

```diff
@@ -1,16 +1,43 @@
 import React from 'react';
+//import { scryRenderedComponentsWithType } from 'react-dom/test-utils';
 import './App.css';
+//import ChatEntry from './components/ChatEntry';
+// import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [likeData, setLikeData] = useState(chatMessages);
+
+  const likeChat = (id) => {
+    setLikeData((likeData) => {
+      return likeData.map((entry) => {
+        if (entry.id === id) {
+          return { ...entry, liked: !entry.liked };
+        } else {
+          return entry;
+        }
+      });
+    });
+  };
+
+  const calcTotalLikes = (likeData) => {
+    return likeData.reduce((total, entry) => {
+      return (total += entry.liked);
+    }, 0);
+  };
```

Comment:
This looks good! One thing that we could do is that remember the 'total += entry.liked' is an assignment which doesn't always work the way we expect in a return statement. You could have removed the "=" and been fine. It's also the syntax that will be more likely to be seen in industry.